tests/integration.rs

```diff
@@ -6728,6 +6748,46 @@ mod common_parallel {
 
         handle_child_output(r, &output);
     }
+
+    #[test]
+    #[cfg(target_arch = "x86_64")]
+    #[cfg(not(feature = "mshv"))]
+    fn test_tpm() {
+        let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+        let guest = Guest::new(Box::new(focal));
+
+        let (mut swtpm_command, swtpm_socket_path) = prepare_swtpm_daemon(&guest.tmp_dir);
+
+        let mut guest_cmd = GuestCommand::new(&guest);
+        guest_cmd
+            .args(["--cpus", "boot=1"])
+            .args(["--memory", "size=512M"])
+            .args(["--kernel", fw_path(FwType::RustHypervisorFirmware).as_str()])
+            .args(["--tpm", &format!("socket={swtpm_socket_path}")])
+            .capture_output()
+            .default_disks()
+            .default_net();
+
+        //start swtpm daemon
+        let mut swtpm_child = swtpm_command.spawn().unwrap();
+        thread::sleep(std::time::Duration::new(2, 0));
```

Comment:
Do you need these 2 seconds? I'm asking because either you really need them and I don't think 2 seconds is safe enough to make it reliable, or you don't need them because Cloud Hypervisor will retry multiple times to connect to the backend for a specific amount of time.

Reply:
I noticed some failures during early development and left this in. I checked this today with 100 runs. They are all pass without a sleep in between. I will drop this in my next update. We can revisit this if we see failures in the future.

Reply:
Well looking at the code, it seems you're lucky and possibly under high load, this is gonna fail. `Tpm::new()` -> `Emulator::new()` -> `SocketDev::init(path)` -> `Socketdev::connect(path)` -> `UnixStream::connect(path)` clearly shows there's no retry mechanism implemented, therefore you should give the `swtpm` daemon enough time to run and listen on the socket path before you run the Cloud Hypervisor binary.
I think it's better to use at least 10 seconds.

Reply:
Gotcha. I thought there was retry login in TPM module for 30 secs. Seems like I dropped the retry logic in TPM module before the final merge.
I will keep the 10 sec delay as you suggested.

I ran some tests last night and did see a failure where cloud-hypervisor failed to connect to swtpm socket.